test/functional/feature_reindex.py

```diff
@@ -19,22 +19,27 @@
 class ReindexTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
-        self.num_nodes = 1
+        self.num_nodes = 2
+        self.extra_args = [
+            [],
+            ["-fastprune"]
+        ]
 
     def reindex(self, justchainstate=False):
         self.generatetoaddress(self.nodes[0], 3, self.nodes[0].get_deterministic_priv_key().address)
         blockcount = self.nodes[0].getblockcount()
-        self.stop_nodes()
-        extra_args = [["-reindex-chainstate" if justchainstate else "-reindex"]]
-        self.start_nodes(extra_args)
+        self.stop_node(0)
```

Comment:
Instead of stopping the node and right after starting it again, you could use `self.restart_node`.

Reply:
ah thanks, fixed. also added an explanation of the test to the comments at the top